README.md

```diff
@@ -82,6 +82,10 @@ module.exports = fp(plugin, {
 })
 ```
 
+#### Base Options
+
+If a plugin has a `options` property, `fastify-plugin` will automatically reference this options object on a `Symbol.for('base-options')` as well. This provides seamless compatibility with [`fastify-autoload`](https://github.com/fastify/fastify-autoload) plugin configuration options shorthand feature.
+
```

Comment:
An example here would be very welcome. I'm not entirely clear what this solves since I don't use `fastify-autoload`.

Reply:
Happy to add an example or maybe we should take the section out entirely since this is purely an internal addition to support a fastify-autoload feature 

But for reference here, the use case is shorthand quick config of an automised module which allows for distributed declarative configuration of plugin libs/modules eg.

```
const helmet = require('fastify-helmet')
helmet.options = { referrerPolicy: true }
module.exports = helmet
```

But this only applies when autoloaded which is why I’m thinking maybe it should be called plugin.autoConfig

Reply:
🤷‍♂ See my feedback on the corresponding autoload PR. I don't understand the utility of this.

Reply:
I think this feature is a shortcut to avoid this:

```
const fp = require('fastify-plugin')
const helmet = require('fastify-helmet')
module.exports = fp((fastify, opts, next) => {
  fastify.register(helmet, { referrerPolicy: true })
  next()
})
```

Am I right?

Reply:
for the most part - yes exactly, it enables a different pattern that's similar to how autoload allows configuration of routes using exported objects.

Reply:
see https://github.com/fastify/fastify-autoload/pull/60#issuecomment-571156995 for a more detailed explanation